Linking C executable bin/mol2inchi fails

Hi

I try to compile `mol2inchi`, but it fails.
Do you have an idea of what happens?

```
[ 93%] Linking C shared library lib/libinchi.so
[ 93%] Built target libinchi
[ 95%] Building C object CMakeFiles/mol2inchi.dir/mol2inchi.c.o
[ 96%] Building C object CMakeFiles/mol2inchi.dir/moreutil.c.o
[ 98%] Building C object CMakeFiles/mol2inchi.dir/home/.../INCHI-1-SRC/INCHI_BASE/src/bcf_s.c.o
[100%] Linking C executable bin/mol2inchi
CMakeFiles/mol2inchi.dir/mol2inchi.c.o: In function `m2i_WorkPool_wait_and_print_all':
/home/.../INCHI-1-SRC/INCHI_API/demos/mol2inchi/src/mol2inchi.c:380: undefined reference to `pthread_join'
CMakeFiles/mol2inchi.dir/mol2inchi.c.o: In function `main':
/home/.../INCHI-1-SRC/INCHI_API/demos/mol2inchi/src/mol2inchi.c:232: undefined reference to `pthread_create'
/home/.../INCHI-1-SRC/INCHI_API/demos/mol2inchi/src/mol2inchi.c:308: undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/mol2inchi.dir/build.make:131: bin/mol2inchi] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/mol2inchi.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

Compiled with
gcc 8.5.0
make 4.2.1
cmake 3.26.5
&
gcc 14.2.0
make 4.4.1
cmake 3.31.3

--- a/INCHI-1-SRC/INCHI_API/demos/mol2inchi/src/CMakeLists.txt
+++ b/INCHI-1-SRC/INCHI_API/demos/mol2inchi/src/CMakeLists.txt
@@ -67,6 +67,10 @@ if(WIN32)
 		COMMAND_EXPAND_LISTS
 	)
 else()
+	set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+	set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+	find_package(Threads REQUIRED)
+	target_link_libraries(mol2inchi PUBLIC Threads::Threads)
 	add_custom_command(TARGET mol2inchi POST_BUILD
 		COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:libinchi> $<TARGET_FILE_DIR:mol2inchi>
 		COMMAND_EXPAND_LISTS

